pymeasure/instruments/agilent/agilentB298x.py

```diff
@@ -0,0 +1,1124 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2025 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+
+from pymeasure.instruments import SCPIMixin, Instrument, Channel
+
+from pymeasure.instruments.validators import (strict_discrete_set,
+                                              strict_range,
+                                              joined_validators
+                                              )
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class Battery(SCPIMixin, Instrument):
```

Comment:
You use that `Battery` as a mixin, never as a standalone instrument, right?

Therefore, you should rename it to `BatteryMixin` and not inherit anything:
```suggestion
class BatteryMixin:
```

Consequently, you do not need the init method either.

Alternatively, you could implement the Battery as a Channel which you add to the respective instruments. See: https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments/channels.html